src/CrudPanel.php

```diff
@@ -226,4 +226,27 @@ public function sort($items)
 
         return $this->{$items};
     }
+
+    /**
+     * Get the Eloquent Model name from the given relation definition string.
+     *
+     * @example For a given string 'company' and a relation between App/Models/User and App/Models/Company, defined by a
+     *          company() method on the user model, the 'App/Models/Company' string will be returned.
+     *
+     * @example For a given string 'company.address' and a relation between App/Models/User, App/Models/Company and
+     *          App/Models/Address defined by a company() method on the user model and an address() method on the
+     *          company model, the 'App/Models/Address' string will be returned.
+     *
+     * @param $relationString String Relation string. A dot notation can be used to chain multiple relations.
+     *
+     * @return string relation model name
+     */
+    private function getRelationModel($relationString)
+    {
+        $result = array_reduce(explode('.', $relationString), function ($obj, $method) {
+            return $obj->$method()->getRelated();
```

Comment:
Ding ding ding! I think ```BadMethodCallException``` is perfect.
![giphy](https://user-images.githubusercontent.com/1032474/29313386-730b2a2a-81c2-11e7-8c16-b8b327b73345.gif)